### PR TITLE
test: (FIR-16861) add system engine integration tests

### PR DESCRIPTION
**Pull request by aymeric-dispa**
- Add system engine integration tests
- Change nightlies to use staging (instead of dev) because dev is unstable
- Add the option to switch env using GHA

### CODE REVIEW

File: driver_integration_test.go [hunk 1]
```diff
@@ -123,3 +127,32 @@ func TestDriverOpenDefaultEngine(t *testing.T) {
...
+	engineNewName := fmt.Sprintf("gosdk_system_engine_test_e_2_%s", suffix)
+
+	db, err := sql.Open("firebolt", dsnSystemEngineMock)
+	if err != nil {
+		t.Errorf("failed unexpectedly with %v", err)
+	}
+	queries := []string{
+		fmt.Sprintf("CREATE DATABASE %s", databaseName),
+		fmt.Sprintf("CREATE ENGINE %s WITH SPEC = 'C1' SCALE = 1", engineName),
+		fmt.Sprintf("ATTACH ENGINE %s TO %s", engineName, databaseName),
```
Comment (ptiurin) [inline driver_integration_test.go:145]: Should we also test DETACH ENGINE? 
Also, SHOW [DATABASES|ENGINES]

File: .github/workflows/integration-tests.yml [hunk 2]
```diff
@@ -2,47 +2,84 @@ name: Integration tests
...
   workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment to run the tests against'
+        type: choice
+        required: true
+        default: 'dev'
+        options:
+          - dev
+          - staging
```
Comment (ptiurin) [inline .github/workflows/integration-tests.yml:13]: I like it. The only downside though is it's not visible which environment the last test run against until you go into the logs.

File: .github/workflows/integration-tests.yml [hunk 3]
```diff
@@ -2,47 +2,84 @@ name: Integration tests
...
+        required: false
 
 jobs:
   integration-tests:
     runs-on: ubuntu-latest
     steps:
       - name: Check out code
-        uses: actions/checkout@v2
-
+        uses: actions/checkout@v3
```
Comment (ptiurin) [inline .github/workflows/integration-tests.yml:43]: Any differences between v2 and v3?

File: .github/workflows/integration-tests.yml [hunk 4]
```diff
@@ -2,47 +2,84 @@ name: Integration tests
...
         with:
           go-version: '1.19.0'
-
+      - name: Determine env variables
+        run: |
+          if [ "${{ github.event.inputs.environment }}" == 'staging' ]; then
+             echo "USERNAME=${{ secrets.FIREBOLT_USERNAME_STAGING }}" >> "$GITHUB_ENV"
+             echo "PASSWORD=${{ secrets.FIREBOLT_PASSWORD_STAGING }}" >> "$GITHUB_ENV"
+             echo "CLIENT_ID=${{ secrets.FIREBOLT_CLIENT_ID_STAGING }}" >> "$GITHUB_ENV"
+             echo "CLIENT_SECRET=${{ secrets.FIREBOLT_CLIENT_SECRET_STAGING }}" >> "$GITHUB_ENV"
```
Comment (ptiurin) [inline .github/workflows/integration-tests.yml:54]: I'm worried this approach might leak the credentials in logs. GH replaces secrets with `***` by default, but what if this environment-agnostic `CLIENT_SECRET` gets dumped in a log?

**Review by ptiurin — COMMENTED** [review]
Just some thoughts. I'd also rename the PR `test:` since you're not actually adding a feature here.

File: .github/workflows/integration-tests.yml (hunk 3 above)
Comment (aymeric-dispa) [reply .github/workflows/integration-tests.yml:43]: v2 is being deprecated (so there are warning poluting the GHA summary) 🤢

File: .github/workflows/integration-tests.yml (hunk 4 above)
Comment (aymeric-dispa) [reply .github/workflows/integration-tests.yml:54]: At the end of the day they are all stored in the environment variables (eg: if you do 'env' in those boxes you'll get the list of all variables in clear, which includes the passwords). As long as we don't decide to logs them we're good 🧷 
 Plus, those are development environments which should not contain any critical data

File: driver_integration_test.go (hunk 1 above)
Comment (aymeric-dispa) [reply driver_integration_test.go:147]: DETACH ENGINE is [being deprecated ](https://docs.firebolt.io/sql-reference/commands/detach-engine.html).
I just added SHOW DATABASES/SHOW ENGINES to the list

File: .github/workflows/integration-tests.yml (hunk 2 above)
Comment (aymeric-dispa) [reply .github/workflows/integration-tests.yml:13]: I just added an extra step to keep the info
image

Comment (ptiurin) [reply .github/workflows/integration-tests.yml:13]: Awesome! No issues then. :)

File: .github/workflows/integration-tests.yml (hunk 4 above)
Comment (ptiurin) [reply .github/workflows/integration-tests.yml:54]: Yeah, I've run `env` and it doesn't print sensitive informtaion https://github.com/firebolt-db/firebolt-go-sdk/actions/runs/3717140023/jobs/6304218056
I disagree on it being only dev/stg though. Those creds are for admin role with ability to create arbitrary number of engines, which can cause a massive AWS bill really quickly. It also can be used to overload the services effectively causing an outage.

File: .github/workflows/integration-tests.yml (hunk 3 above)
Comment (ptiurin) [reply .github/workflows/integration-tests.yml:43]: We should do this for all of our repos. Could you file a ticket to track it please?

File: .github/workflows/integration-tests.yml (hunk 4 above)
Comment (ptiurin) [reply .github/workflows/integration-tests.yml:54]: I've run several other tests trying to output the secrets and looks like they're always obscured so I think we might be reasonably safe here.

**Review by ptiurin — APPROVED** [review]
LGTM

File: driver_integration_test.go [hunk 5]
```diff
@@ -123,3 +127,34 @@ func TestDriverOpenDefaultEngine(t *testing.T) {
...
+		fmt.Sprintf("ALTER DATABASE %s WITH DESCRIPTION = 'GO SDK Integration test'", databaseName),
+		fmt.Sprintf("ALTER ENGINE %s RENAME TO %s", engineName, engineNewName),
+		fmt.Sprintf("START ENGINE %s", engineNewName),
+		fmt.Sprintf("STOP ENGINE %s", engineNewName),
+		fmt.Sprintf("DROP DATABASE %s", databaseName)}
+
+	for _, query := range queries {
+		_, err := db.Query(query)
```
Comment (stepansergeevitch) [inline driver_integration_test.go:155]: Can we also try to read result here is any? So we also test the we support the output?

**Review by stepansergeevitch — APPROVED** [review]
Please resolve a single comment

**Pull request merged**